ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -197,34 +197,44 @@ public static Object toFieldType(final Expression expressionValue) {
   /**
   * Determines the type of command a sql string is and returns a SqlCommand.
   */
-  private static SqlCommand transformToSqlCommand(final String sql) {
+  public static SqlCommand transformToSqlCommand(
+      final String sql, final Map<String, String> variables) {
     final List<String> tokens = Arrays
         .stream(sql.toUpperCase().split(QUOTED_STRING_OR_WHITESPACE))
         .filter(s -> !s.isEmpty())
         .collect(Collectors.toList());
     switch (getStatementType(tokens)) {
       case INSERT_VALUES:
-        return getInsertValuesStatement(sql);
+        return getInsertValuesStatement(sql, variables);
       case CREATE_CONNECTOR:
         return getCreateConnectorStatement(sql);
       case DROP_CONNECTOR:
         return getDropConnectorStatement(sql);
       case STATEMENT:
         return new SqlStatement(sql);
       case SET_PROPERTY:
-        return getSetPropertyCommand(sql);
+        return getSetPropertyCommand(sql, variables);
       case UNSET_PROPERTY:
-        return getUnsetPropertyCommand(sql);
+        return getUnsetPropertyCommand(sql, variables);
+      case DEFINE_VARIABLE:
+        return getDefineVariableCommand(sql);
```

Comment:
This also needs to support variable substitution, right? In case users want to define a variable based on other variables. 

I'm not sure whether it makes sense to support variable substitution for the variable name in addition to for the variable value. It'd be good to check what the ksqlDB CLI's stance on this is, and add support if needed.

Reply:
Updated. Only the variable values can be replaced.